cmd/cluster-agent/app/app.go

```diff
@@ -201,6 +202,20 @@ func start(cmd *cobra.Command, args []string) error {
 		if err := apiserver.StartControllers(ctx); err != nil {
 			log.Errorf("Could not start controllers: %v", err)
 		}
+		// TODO: move rest of the controllers out of the apiserver package
+		orchestratorCtx := orchestrator.ControllerContext{
+			IsLeaderFunc:                 le.IsLeader,
```

Comment:
note: we could reuse this pattern in other controller and remove the interface 🤔